brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/Alien4CloudFacade.java

```diff
@@ -356,6 +359,21 @@ public boolean isDerivedFrom(String nodeId, Alien4CloudApplication toscaApplicat
         return Optional.absent();
     }
 
+    private Set<RelationshipTemplate> findRelationshipsRequirement(String nodeId, Alien4CloudApplication toscaApplication, String requirementId) {
+        Set<RelationshipTemplate> result = MutableSet.of();
+        NodeTemplate node = toscaApplication.getNodeTemplate(nodeId);
+        if (node.getRelationships() != null) {
+            for (Map.Entry<String, RelationshipTemplate> entry : node.getRelationships().entrySet()) {
+                if (entry.getValue().getRequirementName().equals(requirementId)
+                        && (entry.getValue() != null)) {
```

Comment:
If `entry.getValue()` is null the first part of the if block will throw a `NullPointerException`. Swap their order?